src/utils.ts

```diff
@@ -208,3 +221,54 @@ export const normalizePath = (filePath: string): string[] => {
     const absPath = path.resolve(filePath);
     return [path.basename(absPath), absPath];
 };
+
+export const getFileHash = (filename: string): string => {
+    const fileBuffer = fs.readFileSync(filename);
+    const hashSum = crypto.createHash('md5');
+    hashSum.update(fileBuffer);
+    return hashSum.digest('hex');
+};
+
+export const getCachedResults = (context: vscode.ExtensionContext, fileHash: string, logger: Logger): FileScanCacheEntry | undefined => {
+    logger.debug(`Getting cached results for hash ${fileHash}`);
+    validateCacheExpiration(context, logger);
+    const cache: ResultsCacheObject | undefined = context.workspaceState.get(cacheResultsKey);
+    return cache ? cache[fileHash] : undefined;
+};
+
+export const saveCachedResults = (context: vscode.ExtensionContext, fileHash: string, filename: string, results: FailedCheckovCheck[], logger: Logger): void => {
+    logger.debug(`Saving results for file ${filename} (hash: ${fileHash})`);
+    validateCacheExpiration(context, logger);
+
+    const cache: ResultsCacheObject | undefined = context.workspaceState.get(cacheResultsKey);
+    if (cache) {
+        cache[fileHash] = { fileHash, filename, results };
```

Comment:
using the `fileHash` as the key won't result in loads of cache records for each cycle of scan?
The cache is also never cleaned so that might come up as an issue.

Why not use the `filename` as the key as it changes less often? (and then the flow will be to get the record by the filename and then compare the `fileHash`.

Reply:
Good point about clearing old results. I chose the hash because that way it works with scanning a temporary file (when we scan a file that has been edited, but not saved). In my testing, this made it feel much smoother and intuitive than my first version, which did use the filename as the key.

But you are right that it needs to clean up old entries. So I guess the best approach will be to pull cached results for the original file in the case we are scanning a temporary file.